deploy/servctl_utils/kubectl_utils.py

```diff
@@ -92,15 +99,33 @@ def wait_until_pod_is_running(pod_name, deployment_target=None, pod_number=0):
         time.sleep(5)
 
 
-def wait_until_pod_is_ready(pod_name, deployment_target=None):
+def wait_until_pod_is_ready(pod_name, deployment_target=None, pod_number=0):
     """Sleeps until the pod enters "Ready" state"""
 
-    logger.info("waiting for \"%s\" to complete initialization" % pod_name)
-    while not is_pod_ready(pod_name, deployment_target):
+    logger.info("waiting for \"%s\" pod #%s to complete initialization" % (pod_name, pod_number))
+    while not is_pod_ready(pod_name, deployment_target, pod_number=pod_number):
         time.sleep(5)
 
 
-def get_pod_name(pod_name_label, deployment_target=None, pod_number=0):
+def wait_for_resource(resource_name, json_path, expected_status, deployment_target=None, verbose_template='status', resource_type='pod'):
+    """Sleeps until the given resource has the expected status"""
+    while expected_status != _get_resource_status(
+            resource_name, json_path, deployment_target=deployment_target, verbose_template=verbose_template,
+            resource_type=resource_type):
+        time.sleep(5)
+
+
+def wait_for_not_resource(resource_name, json_path, invalid_status, deployment_target=None, verbose_template='status', resource_type='pod'):
+    """Sleeps until the given resource does not have the invalid status"""
+    status = None
+    while not status or status == invalid_status:
+        time.sleep(5)
+        status =  _get_resource_status(
+            resource_name, json_path, deployment_target=deployment_target, verbose_template=verbose_template,
+            resource_type=resource_type)
+
+
+def get_pod_name(pod_name_label, deployment_target=None, pod_number=0, resource_type='pod'):
```

Comment:
Would rename get_pod_name(..) to get_resource_name(..) and make `resource_type` a required arg. 
Otherwise, 
```
pv = get_pod_name(component, deployment_target=deployment_target, resource_type='pv')
```
leads to cognitive dissonance.